app/controllers/spree/paybright_controller.rb

```diff
@@ -52,8 +56,17 @@ def handle_callback_params!
         response_code: paybright_params[:x_gateway_reference],
         amount: paybright_params[:x_amount]
       )
-      @payment.complete!
-      advance_and_complete(@payment.order)
+
+      begin
+        @payment.complete!
+        advance_and_complete(@payment.order)
+      rescue StandardError
+        if @payment.response_code.present?
```

Comment:
What are the semantics/implications of a `response_code` being present nor not?

Reply:
when there is a `response_code`, its the loan id from paybright. so if its present that means that the loan was approved

Reply:
Sounds good, thanks for clarifying!